fe/fe-core/src/main/java/org/apache/doris/nereids/util/DateUtils.java

```diff
@@ -89,7 +89,9 @@ public static DateTimeFormatterBuilder formatBuilder(String pattern) throws Anal
                         break;
                     case 'r': // %r Time, 12-hour (hh:mm:ss followed by AM or PM)
                         builder.appendValue(ChronoField.HOUR_OF_AMPM, 2)
-                                .appendPattern(":mm:ss ")
+                                .appendValue(ChronoField.MINUTE_OF_HOUR, 2)
+                                .appendValue(ChronoField.SECOND_OF_MINUTE, 2)
+                                .appendLiteral(' ')
```

Comment:
Original code exist bug: don't consider 'T'.

BUG is so many, I will fix it by refactor all code about Parsing Date/DateTime